modules/services/mpd-discord-rpc.nix

```
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mpd-discord-rpc;
+  tomlFormat = pkgs.formats.toml { };
+  configFile = tomlFormat.generate "config.toml" cfg.settings;
+in {
+  meta.maintainers = [ maintainers.kranzes ];
+
+  options.services.mpd-discord-rpc = {
+    enable = mkEnableOption "the mpd-discord-rpc service";
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
+      example = {
+        hosts = [ "localhost:6600" ];
+        format = {
+          details = "$title";
+          state = "On $album by $artist";
+        };
+      };
```

```suggestion
      example = literalExpression ''{
        hosts = [ "localhost:6600" ];
        format = {
          details = "$title";
          state = "On $album by $artist";
        };
      }'';
```

Without `literalExpression` the example that is rendered in Home Manager's documentation is on a single line.

---

Also, if there are any settings that are likely to always be set, you could change the type to a freeform module: `types.submodule { freeformType = tomlFormat.type; options.something = mkOption { }; }`

---

done.